### PR TITLE
Remove legacy packaging

### PR DESCRIPTION
**Pull request by prateekchaudhry**
### Summary

This PR removes legacy packaging and references to them from the repo, which are no more being used and have been replaced by newer packaging instead - which all use new build scripts for agent/init
* Remove ```amazon-linux-ami``` which is replaced by ```amazon-linux-ami-integrated```
* Remove ```generic-rpm``` which is replaced by ```generic-rpm-integrated```
* Remove ```generic-deb``` which is replaced by ```generic-deb-integrated```
* Remove ```Init_Makefile``` which is not being used anymore after merging Init with Agent and introducing the new build process. All used targets had been moved to ```Makefile``` to during merge
* Relevant changes in changelog.go referring these packages
* NOTE: Makefile target ```release``` is still being kept around - it is being used for functional tests for PRs, but new target ```release-agent``` is the recommended target
* Remove ```INIT_README.md``` and merge it's content into ```README.md```
* Update ```README.md``` with new build information

### Testing

* Removed legacy packaging - no tests
* Changes in ```scripts/changelog/changelog.go``` tested in personal repo [testChangeLog](https://github.com/prateekchaudhry/amazon-ecs-agent/tree/testChangeLog)
* Readme - no tests
* Ran ```make test, test-init, static-check, static-check-init```

New tests cover the changes: 
No new tests

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by fierlion — APPROVED** [review]
once this change is merged and we're sure there's no un-intended consequences, I would recommend we move all the `-integrated` back to their basic counterparts ie:
amazon-linux-ami-integrated/ -> amazon-linux-ami/

the `integrated` was just to differentiate the new build recipes from the legacy build recipes.

File: README.md [hunk 1]
```diff
@@ -109,6 +109,56 @@ PS C:\> $agentVersion = "v1.20.4"
...
+* `scripts\run-integ-tests.ps1` - Runs all integration tests in the `engine` and `stats` packages
+* `misc\windows-deploy\Install-ECSAgent.ps1` - Install the ECS agent as a Windows service
+* `misc\windows-deploy\amazon-ecs-agent.ps1` - Helper script to set up the host and run the agent as a process
+* `misc\windows-deploy\user-data.ps1` - Sample user-data that can be used with the Windows Server 2016 with Containers
+  AMI to run the agent as a process
+
+
+### Build ECS-Init Package (Linux)
+
+You can also build the ECS-Init packaged as a deb or rpm depending on the linux system you running. Follow instructions at [generic-deb-integrated](https://github.com/aws/amazon-ecs-agent/tree/master/packaging/generic-deb-integrated/debian) and [generic-rpm-integrated](https://github.com/aws/amazon-ecs-agent/tree/master/packaging/generic-rpm-integrated) to build and install ECS Agent packaged with Init deb or rpm packages
```
Comment (mythri-garaga) [inline README.md:160]: nit: ECS-Init package?

**Review by mythri-garaga — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (prateekchaudhry) [reply README.md:160]: I think it can also mean as - ECS-Init can be packaged as a deb or rpm. But I can reword it is unclear.

Comment (mythri-garaga) [reply README.md:160]: I get it now, thank you:)

Comment (prateekchaudhry) [reply README.md:160]: Updating in #3555

**Pull request merged**